packages/sanity/src/core/studio/components/navbar/new-document/NewDocumentButton.tsx

```diff
@@ -46,17 +48,23 @@ export function NewDocumentButton(props: NewDocumentButtonProps) {
 
   const {scheme} = useColorScheme()
   const currentUser = useCurrentUser()
+  const schema = useSchema()
 
   const hasNewDocumentOptions = options.length > 0
   const disabled = !canCreateDocument || !hasNewDocumentOptions
   const placeholder = t('new-document.filter-placeholder')
   const title = t('new-document.title')
   const openDialogAriaLabel = t('new-document.open-dialog-aria-label')
 
-  // Filter options based on search query
+  // Filter options based on search query and document type deprecations
   const filteredOptions = useMemo(
-    () => filterOptions(options, searchQuery, getI18nText),
-    [options, searchQuery, getI18nText],
+    () =>
+      filterOptions(
+        options.filter((option) => !isDeprecatedSchemaType(schema.get(option.schemaType))),
```

Comment:
The filtering of the deprecated schemas does not need to happen on every query change but could happen once like this:

```ts
const nonDeprecatedOptions = useMemo(() => {
  return options.filter((option) => !isDeprecatedSchemaType(schema.get(option.schemaType)))
}, [options, schema])

const filteredOptions = useMemo(
  () => filterOptions(nonDeprecatedOptions, searchQuery, getI18nText),
  [nonDeprecatedOptions, searchQuery, getI18nText],
)
  ```

Reply:
Great point! Thank you!